### ISSUE TITLE
sqlfuff-templater-dbt: Missing whitespace before/after string literal

### ISSUE THREAD

**Issue by kokorin** [sqlfluff/sqlfluff#4010]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

After recent update to sqlfluff 1.4.0 it fails with
```
L:   3 | P:   3 | L048 | Missing whitespace before '1'
L:   5 | P:   3 | L048 | Missing whitespace after '2'
```

DBT model:
```
select
  {%- set x = 42 %}
  '1' as "missing_whitespace_before_1",
  {% set x = 43 -%}
  '2' as "missing_whitespace_after_2",
  {% set x = 44 %}
  '3' as "no_error"
```

### Expected Behaviour

No error should be reported

### Observed Behaviour

The following DBT model 
```
select
  {%- set x = 42 %}
  '1' as "missing_whitespace_1",
  {% set x = 43 -%}
  '2' as "missing_whitespace_2",
  {% set x = 44 %}
  '3' as "no_error"
```
should be interpolated as
```
select
  '1' as "missing_whitespace_1",
  '2' as "missing_whitespace_2",
  '3' as "no_error"
```
which is valid SnowFlake SQL

### How to reproduce

1. Create empty DBT project. 
2. Add DBT model specified above. 
3. Run `sqlfluff lint`

### Dialect

SnowFlake

### Version

sqlfluff 1.4.1 (produces the same error with 1.4.0 but not 1.3.2)
sqlfuff-templater-dbt 1.4.1

```
> dbt --version
Core:
  - installed: 1.3.0
  - latest:    1.3.0 - Up to date!

Plugins:
  - snowflake: 1.3.0 - Up to date!
```

### Configuration

.sqlfluff

```
[sqlfluff]
dialect = snowflake
templater = dbt
exclude_rules = L027,L028,L031,L042,L034,L026

[sqlfluff:templater]
unwrap_wrapped_queries = 1

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:rules:L016]
max_line_length = 120
ignore_comment_lines = 1

[sqlfluff:rules:L010]
capitalisation_policy = lower
```

.sqlfluffignore

```
venv/
target/
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
